server/src/api/v1.py

```diff
@@ -489,6 +491,36 @@ def agents_post(agent_name, json_data):
     return "OK"
 
 
+@v1.post("/agents/status")
+def agents_status_post():
+    """Posts status updates from the agent to the server to be forwarded
+    to TestObserver
+
+    The json sent to this endpoint may contain data such as the following:
+    {
+        "job_state": "<string enum of job states>",
+        "webhook": "<URL as string>",
+        "phases": [
+            {
+            "phase_name": "<string enum of job phases>",
+            "start": "<datetime> | None",
+            "end": "<datetime> | None",
+            "result": <int>,
+            "end_reason": "<string>",
+            },
+        ]
+    }
```

Comment:
Would be great if the structure of the request and response are [schemas](https://github.com/canonical/testflinger/blob/main/server/src/api/schemas.py) rather than a comment. Schemas are exposed on testflinger's [api docs](https://testflinger.canonical.com/docs). And [apiflask](https://apiflask.com/) would provide automatic request validation and serialization